docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,273 @@
+
+## 2. Prepare your environment
+
+Prepare your environment to set up the microk8s cluster:
+
+1. a machine running Ubuntu 18.04 LTS version.
+2. establish a remote session (ssh) onto this machine.
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, please jump on to the [Configuring microk8s](###Configuring-microk8s) section.
+
+## 3. Installing microk8s
+
+You can follow the official documentation on how to [install microk8s](https://microk8s.io/docs/). However, because we were running off of an AWS EC2 instance, we had do the following steps becore installing microk8s:
+
+1. Make sure the machine is up to date and also has the `nfs-common` package installed: `apt update && apt install nfs-common -y`.
+2. We must run configure the instance's list of hosts with localhost. This was because our machine was in fact an AWS EC2 instance, and therefore we must allow the correct resolution of the local microk8s cluster on the kubeconfig. We did so by running `sed -i -e "1s/127.0.0.1 localhost.*/127.0.0.1 localhost $(hostname)/" /etc/hosts`.
+3. Traffic forwarding must be configured on the machine so that the local microk8s cluster pods can reach the internet, again because of using an AWS EC2 instance. This can be done by running `iptables -P FORWARD ACCEPT` (more info on this subject can be [found here](https://microk8s.io/docs/troubleshooting#common-issues).
+
+## 4. Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+   1. `sudo microk8s.enable dns && sudo microk8s.status --wait-ready`
+   2. `sudo microk8s.enable storage && sudo microk8s.status --wait-ready`
+   3. `sudo microk8s.enable ingress &&  sudo microk8s.status --wait-ready`
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster: `microk8s.stop && microk8s.start && microk8s.status --wait-ready`. Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+   1. ```microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s && microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner --timeout=300s && microk8s.kubectl wait -n default --for=condition=Ready pod -l app=default-http-backend --timeout=300s && microk8s.kubectl wait -n default --for=condition=Ready pod -l name=nginx-ingress-microk8s --timeout=300s```
```

Comment:
```suggestion
    
   ```microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s && \
   microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner --timeout=300s && \
   microk8s.kubectl wait -n default --for=condition=Ready pod -l app=default-http-backend --timeout=300s && \
   microk8s.kubectl wait -n default --for=condition=Ready pod -l name=nginx-ingress-microk8s --timeout=300s```
```